File: Lógica de Programação/Linguagem C#/diagonal_negativos.cs
/*

 
Problema "diagonal_negativos"  
Fazer um programa para ler um número inteiro N (máximo = 10) e uma matriz quadrada de ordem N 
contendo números inteiros. Em seguida, mostrar a diagonal principal e a quantidade de valores 
negativos da matriz. 
 
Exemplo: 
Qual a ordem da matriz? 3 
Elemento [0,0]: 5  
Elemento [0,1]: -3  
Elemento [0,2]: 10 
Elemento [1,0]: 15  
Elemento [1,1]: 8  
Elemento [1,2]: 2 
Elemento [2,0]: 7  
Elemento [2,1]: 9  
Elemento [2,2]: -4 
DIAGONAL PRINCIPAL: 
5 8 -4 
QUANTIDADE DE NEGATIVOS = 2 

*/

using System;
using System.Globalization;

class DiagonalNegativos {
    static void Main() {
        
        Console.Write("Qual a ordem da matriz? ");
        int n = int.Parse(Console.ReadLine());

        int[,] mat = new int[10, 10];

        // Leitura da matriz
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                Console.Write("Elemento [" + i + "," + j + "]: ");
                mat[i, j] = int.Parse(Console.ReadLine());
            }
        }

        Console.WriteLine();
        Console.Write("DIAGONAL PRINCIPAL = ");
        for (int i = 0; i < n; i++) {
            Console.Write(mat[i, i] + " ");
        }

        Console.WriteLine();

        int negativos = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (mat[i, j] < 0) {
                    negativos++;
                }
            }
        }

        Console.WriteLine("QUANTIDADE DE NEGATIVOS = " + negativos);
    }
}

File: Lógica de Programação/Linguagem Java/diagonal_negativos.java
/*

Problema "diagonal_negativos"  
Fazer um programa para ler um número inteiro N (máximo = 10) e uma matriz quadrada de ordem N 
contendo números inteiros. Em seguida, mostrar a diagonal principal e a quantidade de valores 
negativos da matriz. 
 
Exemplo: 
Qual a ordem da matriz? 3 
Elemento [0,0]: 5  
Elemento [0,1]: -3  
Elemento [0,2]: 10 
Elemento [1,0]: 15  
Elemento [1,1]: 8  
Elemento [1,2]: 2 
Elemento [2,0]: 7  
Elemento [2,1]: 9  
Elemento [2,2]: -4 
DIAGONAL PRINCIPAL: 
5 8 -4 
QUANTIDADE DE NEGATIVOS = 2 

*/
import java.util.Scanner;
import java.util.Locale;

public class Main {
    public static void main(String[] args) {
        
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);
        
        int[][] mat = new int[10][10];
        int i, j, n, negativos;
        
        negativos = 0;
        
        System.out.printf("Qual a ordem da matriz? ");
        n = sc.nextInt();
        
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
              
              System.out.print("Elemento [ " + i + " , " + j + "]: ");  
              mat[i][j] = sc.nextInt(); 
            }
        }
        
        System.out.println("DIAGONAL PRINCIPAL: ");
        
          for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
              
              if (i == j) {
                  System.out.print(mat[i][j] + " ");
              }
            }
        }
        
        System.out.println();
        
        for (i=0; i<n; i++) {
            for (j=0; j<n; j++) {
              
              if (mat[i][j] < 0) {
                  negativos = negativos + 1;
              }
            }
        }
        
       System.out.println("QUANTIDADE DE NEGATIVOS = " + negativos); 
        
        sc.close();
    }
}
